pqarrow/parquet.go

```diff
@@ -154,17 +154,23 @@ func recordToRows(w dynparquet.ParquetWriter, dcv dynamicColumnVerifier, record
 	return nil
 }
 
-func RecordDynamicCols(record arrow.Record) map[string][]string {
-	dyncols := map[string][]string{}
+func RecordDynamicCols(record arrow.Record) (columns map[string][]string) {
+	dyncols := make(map[string]struct{})
 	for i := 0; i < record.Schema().NumFields(); i++ {
 		af := record.Schema().Field(i)
-		parts := strings.SplitN(af.Name, ".", 2)
-		if len(parts) == 2 { // dynamic column
-			dyncols[parts[0]] = append(dyncols[parts[0]], parts[1])
+		if strings.Contains(af.Name, ".") {
+			dyncols[af.Name] = struct{}{}
 		}
 	}
-
-	return bufutils.Dedupe(dyncols)
```

Comment:
Is dedupe used anywhere now?

Reply:
@thorfour  yes only one place in `ValuesToBuffer`, but when we merge #645

Then I will submit a PR to remove it as well. Because there will  beno any reference to it in `frostdb`.

Reply:
I just rebased to main (In case you are wondering about the forced push)